File: src/main/java/com/mbayou/kick4k/events/type/ChannelSubscriptionGiftsEvent.java
package com.mbayou.kick4k.events.type;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;
import java.util.List;

public class ChannelSubscriptionGiftsEvent extends KickEvent {
    private final EventUser broadcaster;
    private final EventUser gifter;
    private final List<EventUser> giftees;
    private final Instant createdAt;
    private final Instant expiresAt;

    @JsonCreator
    public ChannelSubscriptionGiftsEvent(@JsonProperty("broadcaster") EventUser broadcaster,
                                         @JsonProperty("subscriber") EventUser gifter,
                                         @JsonProperty("giftees") List<EventUser> giftees,
                                         @JsonProperty("created_at") Instant createdAt,
                                         @JsonProperty("expires_at") Instant expiresAt) {
        this.broadcaster = broadcaster;
        this.gifter = gifter;
        this.giftees = giftees;
        this.createdAt = createdAt;
        this.expiresAt = expiresAt;
    }

    public EventUser getBroadcaster() {
        return broadcaster;
    }

    public EventUser getGifter() {
        return gifter;
    }

    public List<EventUser> getGiftees() {
        return giftees;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getExpiresAt() {
        return expiresAt;
    }

    public static String getEventType() {
        return "channel.subscription.gifts";
    }

    public static String getEventVersion() {
        return "1";
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/type/ChannelSubscriptionGiftsEvent.kt
package com.mbayou.kick4k.events.type

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.Instant

data class ChannelSubscriptionGiftsEvent @JsonCreator constructor(
    @JsonProperty("broadcaster") val broadcaster: EventUser?,
    @JsonProperty("subscriber") val gifter: EventUser?,
    @JsonProperty("giftees") val giftees: List<EventUser>?,
    @JsonProperty("created_at") val createdAt: Instant?,
    @JsonProperty("expires_at") val expiresAt: Instant?,
) : KickEvent() {
    companion object {
        @JvmStatic
        fun getEventType(): String = "channel.subscription.gifts"

        @JvmStatic
        fun getEventVersion(): String = "1"
    }
}
